### DISCUSSION TITLE
How crossplane wait dependency resouce created

### DISCUSSION THREAD

**Discussion by vinhnd20** [crossplane/crossplane#5945] [Q&A]
I have a question. For example, now I want to create database resources on my cloud. Then I want to save the Secret including URL, username, password. But there is a problem: the URL only appears after creating the resource database. So is there any mechanism to wait after creating and then save the secret?

**Comment by bobh66** [comment]
You can use Composition Functions with something like `function-go-templating` and check for the existence and status of the database resource in `.observed.resources` before proceeding to extract the information you need from the Secret.

**Reply by hoangndst** [reply]
What about I want to create 2 crossplane provider resources: database and secret_manager. How to make the secret_manager resource wait for database util it provisioned and get the url of the database to create secret using secret_manager crossplane provider resource?

**Reply by vfarcic** [reply]
What would happen if secret_manager (e.g., ESO) cannot retrieve the data (URL) it needs to generate the secret? It will probably fail and try again later. At some moment the database will be operational and secret_manager will be able to get the data it needs to generate the secret. That's a normal behavior in Kubernetes. For example, a Pod want to attach a volume which has not yet been created. It will fail and try again or be in the pending state. Eventually, the volume will be created and Pod will be able to attach it.
